frontend/src/components/PetitionCard.tsx

```diff
@@ -0,0 +1,251 @@
+    </>
+  ));
+}
+
+export default function PetitionCard(props: Props) {
+  const petition = props.petition;
+  const [state, setState] = useState(State.Initial);
+  useEffect(() => {
+    if (state != State.Success && ensureSessionStorage()[petition.slug]) {
+      setState(State.Existed);
+    }
+  });
+
+  const extraQuestions = (petition.extraQuestions || []).map((q) => ({
+    state: useState(""),
+    ...q,
+  }));
+
+  const nameId = "name-field"; // CHANGEME: update to useId in react 18
+  const emailId = "email-field"; // CHANGEME: update to useId in react 18
+
+  const [nameInput, setNameInput] = useState("");
+  const [emailInput, setEmailInput] = useState("");
+
+  if (state == State.Submitting) {
+    return makeBox(
+      props,
+      <div className="self-center py-20">
+        <FaSpinner className="animate-spin" />
+      </div>
+    );
+  }
+  if (state == State.Error) {
+    return makeBox(
+      props,
+      <div className="py-20">Er is iets mis gegaan bij het opsturen.</div>
+    );
+  }
+  if (state == State.Existed) {
+    return makeBox(
+      props,
+      <div className="py-20">Je hebt deze petitie al een keer ingevuld.</div>
+    );
+  }
+  if (state == State.Success) {
+    return makeBox(
+      props,
+      <div className="py-20">
+        Succes! Check je email om je handtekening definitief te maken.
+      </div>
+    );
+  }
+
+  return makeBox(
+    props,
+    <>
+      <h2 className="font-title text-3xl font-bold mb-4">
+        Teken nu voor {petition.title}!
+      </h2>
+      {props.link ? <p>Meer weten? {props.link}</p> : <></>}
+      {petition.signatureCount > 5 ? (
+        <p>
+          Al {petition.signatureCount} mensen hebben deze petitite ondertekend!
+        </p>
+      ) : (
+        <></>
+      )}
+      <div>
+        <label htmlFor={nameId}>Naam:</label>
+      </div>
+      <div>
+        <input
+          id={emailId}
+          type="text"
+          required
+          className="w-full h-8 rounded-md text-black p-2"
+          value={nameInput}
+          onInput={(e) => setNameInput(e.target.value)}
+        ></input>
+      </div>
+      <div>
+        <label htmlFor={emailId}>Email:</label>
+      </div>
+      <div>
+        <input
+          id={emailId}
+          type="email"
+          required
+          className="w-full h-8 rounded-md text-black p-2"
+          value={emailInput}
+          onInput={(e) => setEmailInput(e.target.value)}
+        ></input>
+      </div>
+      {renderExtraQuestions(extraQuestions)}
+      <button
+        type="submit"
+        className="mt-4 font-title text-2xl font-bold border-white border-4 rounded-lg p-1 px-4 w-min"
+      >
+        Teken!
+      </button>
+    </>,
+    (e) => {
+      e.preventDefault();
+      onSubmit(
+        petition,
+        nameInput,
+        emailInput,
+        extraQuestions.map((q) => ({
+          question: q.question,
+          answer: q.state[0],
+        })),
+        setState
+      );
```

Comment:
Gebruik de useCallback-hook om callbacks te definiëren. Is overzichtelijker en ook efficiënter